domain/src/main/java/com/teamagam/gimelgimel/domain/images/SendImageMessageInteractor.java

```diff
@@ -0,0 +1,44 @@
+package com.teamagam.gimelgimel.domain.images;
+
+import com.teamagam.gimelgimel.domain.base.executor.PostExecutionThread;
+import com.teamagam.gimelgimel.domain.base.executor.ThreadExecutor;
+import com.teamagam.gimelgimel.domain.base.interactors.AbstractInteractor;
+import com.teamagam.gimelgimel.domain.images.repository.ImagesRepository;
+import com.teamagam.gimelgimel.domain.messages.entity.MessageImage;
+
+import rx.Observable;
+import rx.Subscriber;
+
+public class SendImageMessageInteractor extends AbstractInteractor {
+
+    private ImagesRepository mImagesRepository;
+    private MessageImage mMessageImage;
+    private String mImagePath;
+
+    public SendImageMessageInteractor(ThreadExecutor threadExecutor, PostExecutionThread postExecutionThread,
+                                      ImagesRepository repository) {
+        super(threadExecutor, postExecutionThread);
+
+        mImagesRepository = repository;
+    }
+
+    public void sendImageMessage(Subscriber subscriber, MessageImage messageImage, String imagePath) {
+        if (isValid(subscriber, messageImage, imagePath)) {
```

Comment:
It's just like what you did in SendMessagePresenter, you checked if the message was null before creating the observable.